### PR TITLE
fix: storybook & webpack 5 successfully build

### PR DESCRIPTION
**Pull request by ScriptedAlchemy**
Issue: Build didn't work 

## What I did
- updated fork ts plugin
- updated webpack to v5 stable
- implemented mjs bandaid loader
- updated loader options API
- kind of fixed cache issues, but needs improvement
- updated webpack virtual modules
- fixed incorrect devtools
## How to test

i ran one example and it seems to build. needs more testing from those who know storybook 

If your answer is yes to any of these, please make sure to include it in your PR.

```
Snapshot Summary
 › 5 snapshots failed from 2 test suites. Inspect your code changes or re-run jest with `-u` to update them.

Test Suites: 5 failed, 121 passed, 126 total
Tests:       9 failed, 1452 passed, 1461 total
Snapshots:   5 failed, 416 passed, 421 total
Time:        75.798 s
Ran all test suites in 7 projects.
```

**there are many references to webpack 4 in here. If I do a global search, everything needs to updated to webpack 5. You can use syncpack to sync it every**

### CODE REVIEW

File: lib/core/src/server/manager/manager-preset.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ export async function managerWebpack(_, options) {
   return createDevConfig(options);
 }
 
-export async function managerEntries(installedAddons, options) {
+export async function manalimitgerEntries(installedAddons, options) {
```
Comment (tnunes) [inline lib/core/src/server/manager/manager-preset.js:8]: Unintended rename?

Comment (ScriptedAlchemy) [reply lib/core/src/server/manager/manager-preset.js:8]: Yes

**Comment by ndelangen** [comment]
@ScriptedAlchemy I tried to merge next into this branch to help you out, but stumbled on quite a few merge conflicts. I tried to resolve them as best as I can. Maybe this help, maybe not:

https://github.com/storybookjs/storybook/tree/pr/ScriptedAlchemy/12964-2

**Comment by ydmitry** [comment]
Any updates on this?

**Comment by phated** [comment]
Webpack 5 support is available in 6.2 Prerelease version now! See the testing instructions at https://gist.github.com/shilman/8856ea1786dcd247139b47b270912324 and if you have any issues, please open a new issue about it so we can take a look!

**Pull request closed without merging**